com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -282,11 +357,18 @@ public void SetClientSynchronizationMode(LoadSceneMode mode)
         /// <summary>
         /// Constructor
         /// </summary>
-        /// <param name="networkManager"></param>
-        internal NetworkSceneManager(NetworkManager networkManager)
+        /// <param name="networkManager">one <see cref="NetworkManager"/> instance per <see cref="NetworkSceneManager"/> instance</param>
+        /// <param name="sceneEventDataPoolSize">maximum <see cref="SceneEventData"/> pool size</param>
+        internal NetworkSceneManager(NetworkManager networkManager, int sceneEventDataPoolSize = DefaultSceneEventDataPoolSize)
         {
+            m_MaxSceneEventDataPoolSize = sceneEventDataPoolSize;
             m_NetworkManager = networkManager;
-            SceneEventData = new SceneEventData(networkManager);
+            SceneEventDataPool = new List<SceneEventData>();
+            for (int i = 0; i < m_MaxSceneEventDataPoolSize; i++)
+            {
+                SceneEventDataPool.Add(new SceneEventData(networkManager));
```

Comment:
I guess I find this odd that we would always warm up the pool with m_MaxSceneEventDataPoolSize allocations of SceneEventData.  Maybe they're not using scene changing at all?  Is the pool growth really that expensive?

Reply:
If a client connects you are using SceneEvents and as such you will be using the SceneEventDataPool.  I am not aware of any network based Unity created game that never loads any scenes during the network session.  Do we think this is possible?